app.rb

```diff
@@ -0,0 +1,131 @@
+require_relative 'rental'
+require_relative 'classroom'
+require_relative 'teacher'
+require_relative 'person'
+require_relative 'book'
+require_relative 'student'
+
+class App
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  def start_console
+    puts 'welcome to our school Library App!'
+    until list_of_options
+      input = gets.chomp
+      if input == '7'
+        puts 'Thank You for using our school Library!'
+        break
+      end
+
+      option input
+    end
+  end
+
+  def list_all_books
+    puts 'Database is empty! Add a book.' if @books.empty?
+    @books.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_all_persons
+    puts 'Database is empty! Add a person.' if @persons.empty?
+    @persons.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, Age: #{person.age}, id: #{person.id}"
+    end
+  end
+
+  def create_person
+    print 'press 1 to create a student, press 2 to create a teacher : '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Try again'
+    end
+  end
+
+  def create_student
+    puts 'create a new student'
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      student = Student.new(age: age, name: name, parent_permission: parent_permission, classroom: @classroom)
+      @persons << student
+      puts 'Student doesn\'t have parent permission, can\'t rent books'
+    when 'y'
+      student = Student.new(age: age, name: name, parent_permission: parent_permission, classroom: @classroom)
+      @persons << student
+      puts 'Student created successfully'
+    end
+  end
+
+  def create_teacher
+    puts 'create a new teacher'
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    @persons << teacher
+    puts 'Teacher created successfully'
+  end
```

Comment:
- [ ] Good job so far, but when entering details for a Teacher, we're expected to enter age, name, and specialization. Currently, your implementation only captures age and specialization and you have left out the name of a Teacher. Kindly add the name of a Teacher. 👍🏽
- [ ] The app crashes as soon as I enter the value for the specialization. Please work on these so a name for a teacher is captured and a teacher is created successfully. 👍🏽

<img width="1304" alt="Screenshot 2023-06-22 at 4 56 45 PM" src="https://github.com/Roland-Ntwali/School-library-app/assets/50941074/744bfa24-36bc-4b02-bba7-a999f1d3a776">